rpcs3/Emu/Cell/SPURecompiler.cpp

```diff
@@ -91,11 +91,60 @@ DECLARE(spu_runtime::tr_interpreter) = []
 
 DECLARE(spu_runtime::g_dispatcher) = []
 {
-	const auto ptr = reinterpret_cast<decltype(spu_runtime::g_dispatcher)>(jit_runtime::alloc(sizeof(spu_function_t), 8, false));
-	ptr->raw() = tr_dispatch;
+	// Allocate 2^20 positions in data area
+	const auto ptr = reinterpret_cast<decltype(spu_runtime::g_dispatcher)>(jit_runtime::alloc(8 * 65536 * 16, 8, false));
+
+	for (u32 i = 0; i < 65536 * 16; i++)
+	{
+		ptr[i].raw() = tr_dispatch;
+	}
```

Comment:
Micro-optimization - using word size variable (in this case `size_t`) **may** produce more efficient code:
https://www.viva64.com/en/a/0050/

This case is different due to lack of `- 1` in it but it's still worth going for IMHO.

Reply:
I wanted to explain this: the compiler must not optimize unsigned archimatics if theres a potential for an overflow.
This means it must do the addition in a 32-bits op instead of 64-bits, limiting the optimization.
Using an s32 should solve this as signed overflow is a UB and compilers are allow to optimize this.

Reply:
Is there really any potential for an overflow here? The range of values is known, and `65536 * 16` will naturally fit in both 32-bit and 64-bit registers.